fuzz/fuzzers/fuzzer_script_exr.rs

```diff
@@ -1,28 +1,41 @@
 #![no_main]
-#[macro_use] extern crate libfuzzer_sys;
+#[macro_use]
+extern crate libfuzzer_sys;
 extern crate image;
 
-use std::io::Cursor;
-use image::ImageResult;
 use image::codecs::openexr::*;
-use std::io::Seek;
-use std::convert::TryFrom;
+use image::ColorType;
 use image::ImageDecoder;
 use image::ImageEncoder;
-use image::ColorType;
-use std::io::Write;
+use image::ImageResult;
+use std::convert::TryFrom;
+use std::io::Cursor;
 use std::io::Read;
+use std::io::Seek;
+use std::io::SeekFrom;
+use std::io::Write;
 
 // "just dont panic"
 fn roundtrip(bytes: &[u8]) -> ImageResult<()> {
-
     /// Read the file from the specified path into an `Rgba32FImage`.
     // TODO this method should probably already exist in the main image crate
-    fn read_as_rgba_byte_image(read: impl Read + Seek) -> ImageResult<(u32,u32,Vec<u8>)> {
+    fn read_as_rgba_byte_image(read: &mut (impl Read + Seek)) -> ImageResult<(u32, u32, Vec<u8>)> {
+        let current_pos = read.stream_position()?;
+        let total_size = read.seek(SeekFrom::End(0))?;
+        let _ = read.seek(SeekFrom::Start(current_pos))?;
+
         let decoder = OpenExrDecoder::with_alpha_preference(read, Some(true))?;
         let (width, height) = decoder.dimensions();
 
-        let mut buffer = vec![0; usize::try_from(decoder.total_bytes()).unwrap()];
+        let decoded_size = usize::try_from(decoder.total_bytes()).unwrap();
+        if total_size - current_pos < decoded_size as u64 {
```

Comment:
I don't think this is the logic we want. The EXR format supports compression, so the decoded size is _usually_ going to be larger than the file size. Instead, for fuzzing purposes we could just impose a maximum size of say 256 MB. 

(As a smaller point, we should probably check decoded_size before calling usize::try_from(...).unwrap() so that we don't fail there on 32-bit systems)

Reply:
If we upper bound the decoded size against anything, then comparing against the limits configured for the reader would also make sense (i.e. `Limits::max_alloc). We *are* doing another immediate allocation afterall.

Reply:
If it's unclear, I'm suggesting that properly supporting `<OpenExrDecoder as ImageDecoder>::set_limits` and then using that function to configure the limit would be a much more thorough fix. That way, we get better feedback if all allocations are accounted for or if the fuzzer manages to find additional ones inside the decoder.

Reply:
> I don't think this is the logic we want. The EXR format supports compression, so the decoded size is _usually_ going to be larger than the file size. Instead, for fuzzing purposes we could just impose a maximum size of say 256 MB.
> 
> (As a smaller point, we should probably check decoded_size before calling usize::try_from(...).unwrap() so that we don't fail there on 32-bit systems)

Thanks, I didn't know that there is a compression:). I changed fix on what you offer.

Reply:
> If it's unclear, I'm suggesting that properly supporting `<OpenExrDecoder as ImageDecoder>::set_limits` and then using that function to configure the limit would be a much more thorough fix. That way, we get better feedback if all allocations are accounted for or if the fuzzer manages to find additional ones inside the decoder.

Sounds pretty good, I need a little bit more time to look at code, maybe find some examples for this

Reply:
I've provided an implementation for `OpenExrDecoder::set_limits` function. In this function I check some common cases and a case with `total_bytes()`. Maybe we need to save `Limits` in `Decoder`, and check them sometimes. But I don't familiar with exr decoding for now:). What do I think?

Reply:
@HeroicKatora the unbounded allocation that's being fixed here is actually in the fuzz harness itself, which set_limits isn't designed to address